node/config/types.go

```diff
@@ -279,6 +279,50 @@ type ContractDealsConfig struct {
 	From string
 }
 
+type IndexProviderConfig struct {
+	// Enable set whether to enable indexing announcement to the network and expose endpoints that
+	// allow indexer nodes to process announcements. Enabled by default.
+	Enable bool
+
+	// EntriesCacheCapacity sets the maximum capacity to use for caching the indexing advertisement
+	// entries. Defaults to 1024 if not specified. The cache is evicted using LRU policy. The
+	// maximum storage used by the cache is a factor of EntriesCacheCapacity, EntriesChunkSize and
+	// the length of multihashes being advertised. For example, advertising 128-bit long multihashes
+	// with the default EntriesCacheCapacity, and EntriesChunkSize means the cache size can grow to
+	// 256MiB when full.
+	EntriesCacheCapacity int
+
+	// EntriesChunkSize sets the maximum number of multihashes to include in a single entries chunk.
+	// Defaults to 16384 if not specified. Note that chunks are chained together for indexing
+	// advertisements that include more multihashes than the configured EntriesChunkSize.
+	EntriesChunkSize int
+
+	// TopicName sets the topic name on which the changes to the advertised content are announced.
+	// If not explicitly specified, the topic name is automatically inferred from the network name
+	// in following format: '/indexer/ingest/<network-name>'
+	// Defaults to empty, which implies the topic name is inferred from network name.
+	TopicName string
+
+	// PurgeCacheOnStart sets whether to clear any cached entries chunks when the provider engine
+	// starts. By default, the cache is rehydrated from previously cached entries stored in
+	// datastore if any is present.
+	PurgeCacheOnStart bool
+
+	HttpEndpoint IndexProviderHttpEndpointConfig
+}
+
+type IndexProviderHttpEndpointConfig struct {
+	// Set the public IPv4 endpoint for the index provider listener.
+	// eg "82.129.73.111"
+	// This is usually the same as the public IP for the boost node.
+	// If PublicIP is the empty string, requests are served over graphsync
+	// instead.
+	PublicIP string
```

Comment:
To be clear this ends up being included in the HTTP announce messages as _multiaddr_, which can be dns name, etc. as well as an IPv4. We already see SPs that are using dns names in their setup so forcing this to be an IP might come back to haunt us for SPs that have name servers setup and use DNS for routing.

[This](https://github.com/ipni/go-libipni/blob/afe2d8ea45b86c2a22f756ee521741c8f99675e5/maurl/convert.go#L106) utility function might be handy to work around this.